packages/web3/src/contract/ralph.ts

```diff
@@ -466,7 +488,7 @@ export function buildScriptByteCode(
 ): string {
   const allFields = flattenFields(fields, fieldsSig.names, fieldsSig.types, fieldsSig.isMutable, structs)
   return bytecodeTemplate.replace(scriptFieldRegex, (_, fieldIndex: string) => {
-    const field = allFields[`${fieldIndex}`]
+    const field = allFields[parseInt(fieldIndex)]
```

Comment:
Question: why do we need to change this?

Reply:
reverted now